common/flatpak-dir.c

```diff
@@ -3874,52 +3875,37 @@ copy_remote_config (GKeyFile *config,
     }
 }
 
-static GHashTable *
-_flatpak_dir_find_new_flatpakrepos (FlatpakDir *self, OstreeRepo *repo)
+static void
+_flatpak_dir_scan_new_flatpakrepos (const char          *dir_str,
+                                    GHashTable         **flatpakrepos,
+                                    const char * const  *remotes)
 {
-  g_autoptr(GHashTable) flatpakrepos = NULL;
-  g_autoptr(GFile) conf_dir = NULL;
+  g_autoptr(GFile) dir = NULL;
   g_autoptr(GFileEnumerator) dir_enum = NULL;
   g_autoptr(GError) my_error = NULL;
-  g_autofree char *config_dir = NULL;
-  g_auto(GStrv) remotes = NULL;
-  g_auto(GStrv) applied_remotes = NULL;
 
-  g_assert (repo != NULL);
-
-  /* Predefined remotes only applies for the default system installation */
-  if (self->user ||
-      (self->extra_data != NULL &&
-       strcmp (self->extra_data->id, SYSTEM_DIR_DEFAULT_ID) != 0))
-    return NULL;
+  g_return_if_fail (dir_str != NULL);
+  g_return_if_fail (flatpakrepos != NULL);
 
-  config_dir = g_strdup_printf ("%s/%s",
-                                get_config_dir_location (),
-                                SYSCONF_REMOTES_DIR);
-  conf_dir = g_file_new_for_path (config_dir);
-  dir_enum = g_file_enumerate_children (conf_dir,
+  dir = g_file_new_for_path (dir_str);
+  dir_enum = g_file_enumerate_children (dir,
                                         G_FILE_ATTRIBUTE_STANDARD_NAME "," G_FILE_ATTRIBUTE_STANDARD_TYPE,
                                         G_FILE_QUERY_INFO_NONE,
                                         NULL, &my_error);
   if (my_error != NULL)
-    return NULL;
-
-  remotes = ostree_repo_remote_list (repo, NULL);
-  applied_remotes = g_key_file_get_string_list (ostree_repo_get_config (repo),
-                                                "core", "xa.applied-remotes", NULL, NULL);
+    return;
 
   while (TRUE)
     {
-      GFileInfo *file_info;
-      GFile *path;
+      g_autoptr(GFileInfo) file_info = NULL;
```

Comment:
according to the doc, it is transfer-none.
> Another crucial difference is that the references for out_info and out_child are owned by direnum (they are cached as hidden properties). You must not unref them in your own code. This makes memory management significantly easier for C code in combination with loops.

So this is just incorrect. `path` is unused, but not leaked.

Reply:
Aha, you're right! 

I guess there's a bit of a glitch in the documentation. https://gnome.pages.gitlab.gnome.org/gtk/gio/method.FileEnumerator.iterate.html states: "the data is owned by the caller of the function", but has no reference to `out` as in the gio docs. So basically states the opposite to reality? Or am I just understanding it completely wrong?

Reply:
DevHelp which show the old doc:

![image](https://github.com/flatpak/flatpak/assets/114441/f2db58d0-dfbf-46dc-af22-efc9bd567ff1)

So there is an actual problem here.

At least in the wording. I'd file an issue.

Reply:
Seems like it was already [fixed](https://gitlab.gnome.org/GNOME/gi-docgen/-/commit/5d360a1a590e3d6d46c2cef72a55ed4d738d931c), but only in 2023.2 gi-docgen release, and the website is using 2023.1, will try to update